diff --git a/CMakeLists.txt b/CMakeLists.txt
index efe857df..a08e8f8a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,10 +23,10 @@
 # LIABILITY, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
 # OF THIS SOFTWARE.
 
-cmake_minimum_required(VERSION 2.8.11) # b/c of use of BUILD_INTERFACE generator expression
+cmake_minimum_required(VERSION 3.5) # b/c of use of BUILD_INTERFACE generator expression
 
-# Default policy is from 2.8.9
-cmake_policy(VERSION 2.8.9)
+# Default policy is from 3.5
+cmake_policy(VERSION 3.5)
 # Set MacOSX @rpath usage globally.
 if (POLICY CMP0020)
   cmake_policy(SET CMP0020 NEW)
@@ -309,28 +309,32 @@ elseif(SIZEOF_UNSIGNED_LONG EQUAL 4)
   set(TIFF_UINT32_FORMAT "%lu")
 endif()
 
-if(SIZEOF_SIGNED_LONG EQUAL 8)
-  set(TIFF_INT64_T "signed long")
-  set(TIFF_INT64_FORMAT "%ld")
-elseif(SIZEOF_SIGNED_LONG_LONG EQUAL 8)
+# Open 3d Engine change: always choose the 'longest long'
+# so that its compatible with other code choosing long long
+# to represent i64 since that is the same size across
+# more platforms.
+if(SIZEOF_SIGNED_LONG_LONG EQUAL 8)
   set(TIFF_INT64_T "signed long long")
   if (MINGW)
     set(TIFF_INT64_FORMAT "%I64d")
   else()
     set(TIFF_INT64_FORMAT "%lld")
   endif()
+elseif(SIZEOF_SIGNED_LONG EQUAL 8)
+  set(TIFF_INT64_T "signed long")
+  set(TIFF_INT64_FORMAT "%ld")
 endif()
 
-if(SIZEOF_UNSIGNED_LONG EQUAL 8)
-  set(TIFF_UINT64_T "unsigned long")
-  set(TIFF_UINT64_FORMAT "%lu")
-elseif(SIZEOF_UNSIGNED_LONG_LONG EQUAL 8)
-  set(TIFF_UINT64_T "unsigned long long")
+if(SIZEOF_UNSIGNED_LONG_LONG EQUAL 8)
+set(TIFF_UINT64_T "unsigned long long")
   if (MINGW)
     set(TIFF_UINT64_FORMAT "%I64u")
   else()
     set(TIFF_UINT64_FORMAT "%llu")
   endif()
+elseif(SIZEOF_UNSIGNED_LONG EQUAL 8)
+  set(TIFF_UINT64_T "unsigned long")
+  set(TIFF_UINT64_FORMAT "%lu")
 endif()
 
 if(SIZEOF_UNSIGNED_INT EQUAL SIZEOF_SIZE_T)
