diff --unified --recursive --no-dereference a/GameLift-Cpp-ServerSDK-5.0.0/gamelift-server-sdk/include/aws/gamelift/internal/model/request/UpdatePlayerSessionCreationPolicyRequest.h b/GameLift-Cpp-ServerSDK-5.0.0/gamelift-server-sdk/include/aws/gamelift/internal/model/request/UpdatePlayerSessionCreationPolicyRequest.h
--- a/GameLift-SDK-Release-5.0.0/GameLift-SDK-Release-5.0.0/GameLift-Cpp-ServerSDK-5.0.0/gamelift-server-sdk/include/aws/gamelift/internal/model/request/UpdatePlayerSessionCreationPolicyRequest.h	2023-02-08 13:28:26.000000000 -0800
+++ b/GameLift-SDK-Release-5.0.0/GameLift-SDK-Release-5.0.0/GameLift-Cpp-ServerSDK-5.0.0/gamelift-server-sdk/include/aws/gamelift/internal/model/request/UpdatePlayerSessionCreationPolicyRequest.h	2023-05-08 07:52:32.249104692 -0700
@@ -114,7 +114,7 @@
         static constexpr const char* NOT_SET = "NOT_SET";
 
         std::string m_gameSessionId;
-        WebSocketPlayerSessionCreationPolicy m_playerSessionCreationPolicy;
+        WebSocketPlayerSessionCreationPolicy m_playerSessionCreationPolicy = WebSocketPlayerSessionCreationPolicy::NOT_SET;
     };
 } // namespace Internal
 } // namespace GameLift
