diff --git a/src/Camera.cc b/src/Camera.cc
index 3d7cd39c..8d33e1d6 100644
--- a/src/Camera.cc
+++ b/src/Camera.cc
@@ -1123,9 +1123,9 @@ void Camera::SetLensIntrinsicsSkew(double _s)
 /////////////////////////////////////////////////
 std::string Camera::ConvertPixelFormat(PixelFormatType _type)
 {
-  unsigned int index = static_cast<int>(_type);
+  unsigned int index = static_cast<unsigned int>(_type);
   if (index < kPixelFormatNames.size())
-    return std::string(kPixelFormatNames[static_cast<int>(_type)]);
+    return std::string(kPixelFormatNames[index]);
 
   return std::string(kPixelFormatNames[0]);
 }
diff --git a/src/ParticleEmitter.cc b/src/ParticleEmitter.cc
index b0d07a20..dbb21788 100644
--- a/src/ParticleEmitter.cc
+++ b/src/ParticleEmitter.cc
@@ -267,7 +267,7 @@ bool ParticleEmitter::SetType(const std::string &_typeStr)
 /////////////////////////////////////////////////
 std::string ParticleEmitter::TypeStr() const
 {
-  size_t index = static_cast<int>(this->dataPtr->type);
+  size_t index = static_cast<size_t>(this->dataPtr->type);
   if (index < kEmitterTypeStrs.size())
     return std::string(kEmitterTypeStrs[index]);
   return "point";
diff --git a/src/Sensor.cc b/src/Sensor.cc
index 8f3b1205..86d708e4 100644
--- a/src/Sensor.cc
+++ b/src/Sensor.cc
@@ -663,7 +663,7 @@ void Sensor::SetUpdateRate(double _hz)
 /////////////////////////////////////////////////
 std::string Sensor::TypeStr() const
 {
-  size_t index = static_cast<int>(this->dataPtr->type);
+  size_t index = static_cast<size_t>(this->dataPtr->type);
   if (index > 0 && index < kSensorTypeStrs.size())
     return std::string(kSensorTypeStrs[index]);
   return "none";
