From 3176985bc5136e5b7658ab22d85e406c5da4bd82 Mon Sep 17 00:00:00 2001
Date: Fri, 11 Mar 2022 13:25:09 -0800
Subject: [PATCH] Enable stdio on iOS

---
 ports/openssl/unix/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ports/openssl/unix/CMakeLists.txt b/ports/openssl/unix/CMakeLists.txt
index 7bf45e27b..8cdb65fc3 100644
--- a/ports/openssl/unix/CMakeLists.txt
+++ b/ports/openssl/unix/CMakeLists.txt
@@ -23,7 +23,7 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "iOS")
         message(FATAL_ERROR "Unknown iOS target architecture: ${VCPKG_TARGET_ARCHITECTURE}")
     endif()
     # disable that makes linkage error (e.g. require stderr usage)
-    list(APPEND DISABLES no-stdio no-ui no-asm)
+    list(APPEND DISABLES no-ui no-asm)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     if(VCPKG_TARGET_ARCHITECTURE MATCHES "arm64")
         set(PLATFORM darwin64-arm64-cc)
-- 
2.34.0.windows.1

