diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ed55aad7..87a302d85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -405,7 +405,7 @@ if (WIN32)
 endif ()
 
 if (BUILD_SHARED_LIBS)
-  set_target_properties(freetype PROPERTIES
+set_target_properties(freetype PROPERTIES
     VERSION ${LIBRARY_VERSION}
     SOVERSION ${LIBRARY_SOVERSION})
 endif ()
diff --git a/include/ft2build.h b/include/ft2build.h
index b4fd1f8c3..56c261b10 100644
--- a/include/ft2build.h
+++ b/include/ft2build.h
@@ -34,6 +34,10 @@
 #ifndef FT2BUILD_H_
 #define FT2BUILD_H_
 
+#ifndef FT2_BUILD_LIBRARY
+#define FT2_BUILD_LIBRARY
+#endif
+
 #include <freetype/config/ftheader.h>
 
 #endif /* FT2BUILD_H_ */
