diff --git a/crt/aws-crt-cpp/include/aws/crt/io/TlsOptions.h b/crt/aws-crt-cpp/include/aws/crt/io/TlsOptions.h
index 7205d8f..fa6a909 100644
--- a/crt/aws-crt-cpp/include/aws/crt/io/TlsOptions.h
+++ b/crt/aws-crt-cpp/include/aws/crt/io/TlsOptions.h
@@ -89,6 +89,7 @@ namespace Aws
                     const char *pkcs12_pwd,
                     Allocator *allocator = g_allocator) noexcept;
 
+#if !defined(AWS_OS_IOS)
                 /**
                  * By default the certificates and private keys are stored in the default keychain
                  * of the account of the process. If you instead wish to provide your own keychain
@@ -96,6 +97,8 @@ namespace Aws
                  * NOTE: The password of your keychain must be empty.
                  */
                 bool SetKeychainPath(ByteCursor &keychain_path) noexcept;
+#endif /* !AWS_OS_IOS */
+
 #endif
 
 #ifdef _WIN32
diff --git a/crt/aws-crt-cpp/source/io/TlsOptions.cpp b/crt/aws-crt-cpp/source/io/TlsOptions.cpp
index b9305e5..f32f873 100644
--- a/crt/aws-crt-cpp/source/io/TlsOptions.cpp
+++ b/crt/aws-crt-cpp/source/io/TlsOptions.cpp
@@ -106,11 +106,14 @@ namespace Aws
                 return ctxOptions;
             }
 
+#if !defined(AWS_OS_IOS)
             bool TlsContextOptions::SetKeychainPath(ByteCursor &keychain_path) noexcept
             {
                 AWS_ASSERT(m_isInit);
                 return aws_tls_ctx_options_set_keychain_path(&m_options, &keychain_path) == 0;
             }
+#endif /* !AWS_OS_IOS */
+
 #endif /* AWS_OS_APPLE */
 
 #ifdef _WIN32
