diff --git a/aws-cpp-sdk-core/include/aws/core/auth/AWSAuthSigner.h b/aws-cpp-sdk-core/include/aws/core/auth/AWSAuthSigner.h
index 8061ee1e25..591d65555a 100644
--- a/aws-cpp-sdk-core/include/aws/core/auth/AWSAuthSigner.h
+++ b/aws-cpp-sdk-core/include/aws/core/auth/AWSAuthSigner.h
@@ -399,6 +399,21 @@ namespace Aws
              */
             bool SignRequest(Aws::Http::HttpRequest&) const override { return true; }
 
+            /**
+             * Do nothing
+             */
+            bool SignRequest(Aws::Http::HttpRequest&, bool) const override { return true; }
+
+            /**
+             * Do nothing
+             */
+            bool SignRequest(Aws::Http::HttpRequest&, const char*, bool) const override { return true; }
+
+            /**
+             * Do nothing
+             */
+            bool SignRequest(Aws::Http::HttpRequest&, const char*, const char*, bool) const override { return true; }
+
             /**
              * Do nothing
              */
